docs/source/finetuning.rst

```diff
@@ -0,0 +1,291 @@
+model optimization flow by leveraging our quantization and sparsity
+techniques integrated into our partner frameworks. This is part 2 of 3
+such tutorials showcasing this end-to-end flow, focusing on the
+fine-tuning step.
+
+.. image:: ../static/e2e_flow_part2.png
+
+Fine-tuning is an important step for adapting your pre-trained model
+to more domain-specific data. In this tutorial, we demonstrate 3 model
+optimization techniques that can be applied to your model during fine-tuning:
+
+1. **Quantization-Aware Training (QAT)**, for adapting your model to
+quantization numerics during fine-tuning, with the goal of mitigating
+quantization degradations in your fine-tuned model when it is quantized
+eventually, e.g. in the serving step. Check out `our blog <https://pytorch.org/blog/quantization-aware-training/>`__
+for more details!
+
+2. **Quantized Low-Rank Adaptation (QLoRA)**, for reducing the resource
+requirement of fine-tuning by introducing small, trainable low-rank
+matrices and freezing the original pre-trained checkpoint, a type of
+Parameter-Efficient Fine-Tuning (PEFT). Please refer to the `original
+paper <https://arxiv.org/pdf/2305.14314>`__ for more details.
+
+3. **Float8 Quantized Fine-tuning**, for speeding up fine-tuning by
+dynamically quantizing high precision weights and activations to float8,
+similar to `pre-training in float8 <pretraining.html>`__.
+
+
+Quantization-Aware Training (QAT)
+##################################
+
+The goal of Quantization-Aware Training is to adapt the model to
+quantization numerics during training or fine-tuning, so as to mitigate
+the inevitable quantization degradation when the model is actually
+quantized eventually, presumably during the serving step after fine-tuning.
+TorchAO's QAT support has been used successfully for the recent release of
+the `Llama-3.2 quantized 1B/3B <https://ai.meta.com/blog/meta-llama-quantized-lightweight-models/>`__
+and the `LlamaGuard-3-8B <https://github.com/meta-llama/PurpleLlama/blob/main/Llama-Guard3/8B/MODEL_CARD.md>`__ models to improve the quality of the quantized models.
+
+TorchAO's QAT support involves two separate steps: prepare and convert.
+The prepare step "fake" quantizes activations and/or weights during
+training, which means, the high precision values (e.g. bf16) are mapped
+to their corresponding quantized values *without* actually casting them
+to the target lower precision dtype (e.g. int4). The convert step,
+applied after training, replaces "fake" quantization operations in the
+model with "real" quantization that does perform the dtype casting:
+
+.. image:: ../../torchao/quantization/qat/images/qat_diagram.png
+
+There are multiple options for using TorchAO's QAT for fine-tuning:
+1. Directly use our QAT APIs with your own training loop
+2. Use our integration with `TorchTune <https://github.com/pytorch/torchtune>`__
+3. Use our integratino with `Axolotl <https://github.com/axolotl-ai-cloud/axolotl>`__
+
+
+Option 1: TorchAO QAT API
```

Comment:
when do we want people use this?

Reply:
I think if people have their own fine-tuning frameworks they want to use, e.g. the Llama-3.2 1B/3B quantized release did not use torchtune or axolotl